functions-and-operators/encryption-and-compression-functions.md

```diff
@@ -12,30 +12,422 @@ TiDB supports most of the [encryption and compression functions](https://dev.mys
+8 rows in set (0.01 sec)
+```
+
+We first check the password strength of an empty string, which returns 0.
+
+```sql
+SELECT VALIDATE_PASSWORD_STRENGTH('');
+```
+
+```
++--------------------------------+
+| VALIDATE_PASSWORD_STRENGTH('') |
++--------------------------------+
+|                              0 |
++--------------------------------+
+1 row in set (0.00 sec)
+```
+
+Then we check the password strength of an short string, which returns 25.
+
+```sql
+SELECT VALIDATE_PASSWORD_STRENGTH('abcdef');
+```
+
+```
++--------------------------------------+
+| VALIDATE_PASSWORD_STRENGTH('abcdef') |
++--------------------------------------+
+|                                   25 |
++--------------------------------------+
+1 row in set (0.00 sec)
+```
+
+Then we check the password strength of an longer string, which returns 50. This is now more characters than we have configured with [`validate_password.length`](/system-variables.md#validate_passwordlength-new-in-v650).
+
+```sql
+SELECT VALIDATE_PASSWORD_STRENGTH('abcdefghi');
+```
+
+```
++-----------------------------------------+
+| VALIDATE_PASSWORD_STRENGTH('abcdefghi') |
++-----------------------------------------+
+|                                      50 |
++-----------------------------------------+
+1 row in set (0.00 sec)
+```
+
+Now we add an upper-case character. This still returns 50.
+
+```sql
+SELECT VALIDATE_PASSWORD_STRENGTH('Abcdefghi');
+```
+
+```
++-----------------------------------------+
+| VALIDATE_PASSWORD_STRENGTH('Abcdefghi') |
++-----------------------------------------+
+|                                      50 |
++-----------------------------------------+
+1 row in set (0.01 sec)
+```
+
+Now we add numbers. This still returns 50.
+
+```sql
+SELECT VALIDATE_PASSWORD_STRENGTH('Abcdefghi123');
+```
+
+```
++--------------------------------------------+
+| VALIDATE_PASSWORD_STRENGTH('Abcdefghi123') |
++--------------------------------------------+
+|                                         50 |
++--------------------------------------------+
+1 row in set (0.00 sec)
+```
+
+Now we add special characters and we finally get to 100, which means this is a strong password.
+
+```sql
+SELECT VALIDATE_PASSWORD_STRENGTH('Abcdefghi123%$#');
+```
+
+```
++-----------------------------------------------+
+| VALIDATE_PASSWORD_STRENGTH('Abcdefghi123%$#') |
++-----------------------------------------------+
+|                                           100 |
++-----------------------------------------------+
+1 row in set (0.00 sec)
+```
 
 ## Unsupported functions
 
-* `DES_DECRYPT()`, `DES_ENCRYPT()`, `OLD_PASSWORD()`, `ENCRYPT()`: these functions were deprecated in MySQL 5.7 and removed in 8.0.
-* Functions only available in MySQL Enterprise [Issue #2632](https://github.com/pingcap/tidb/issues/2632).
+* TiDB does not support the functions only available in MySQL Enterprise [Issue #2632](https://github.com/pingcap/tidb/issues/2632).
```

Comment:
Should we list them?

Reply:
I don't think that's needed.

They are listed on https://dev.mysql.com/doc/refman/8.4/en/enterprise-encryption-functions.html